### PR TITLE
FastLED documentation update for esphome/esphome#4257

### PR DESCRIPTION
**Pull request by bibach**
## Description:

Update FastLED documentation to reflect new, unified _fastled_ component and indicate that the existing, separate components are now deprecated.

**Related issue (if applicable):** 

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4257

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/light/fastled.rst [hunk 1]
```diff
@@ -26,11 +26,122 @@ FastLED Light
...
+
+.. figure:: images/fastled_spi-ui.png
+    :align: center
+    :width: 60.0%
+
+.. code-block:: yaml
+
+    # Example Clockless configuration entry
+    light:
+      - platform: fastled_clockless
```
Comment (nielsnl68) [inline components/light/fastled.rst]:
```suggestion
      - platform: fastled
```

File: components/light/fastled.rst [hunk 2]
```diff
@@ -26,11 +26,122 @@ FastLED Light
...
+        pin: GPIO23
+        num_leds: 60
+        rgb_order: BRG
+        name: "FastLED WS2811 Light"
+
+.. code-block:: yaml
+
+    # Example SPI configuration entry
+    light:
+      - platform: fastled_spi
```
Comment (nielsnl68) [inline components/light/fastled.rst]:
```suggestion
      - platform: fastled
```

File: components/light/fastled.rst [hunk 3]
```diff
@@ -107,6 +218,11 @@ Supported Chipsets
 SPI
 ---
 
+.. warning::
+
+    The ``fastled_spi`` light platform is deprecated.  Use the new,
+    unified ``fastled`` light platform, instead.
+
 The ``fastled_spi`` light platform allows you to create RGB lights
 in ESPHome for a :ref:`number of supported chipsets <fastled_spi-chipsets>`.
```
Comment (nielsnl68) [inline components/light/fastled.rst]:
```suggestion
```

My suggestion is moving the warning to the top and remove this sections.

File: components/light/fastled.rst [hunk 4]
```diff
@@ -26,11 +26,122 @@ FastLED Light
...
+FastLED
+-------
+
+The ``fastled`` light platform allows you to create RGB lights in ESPHome
+for a :ref:`number of supported chipsets <fastled-chipsets>`.
+
+Both clockless and SPI chips are supported, with the former only requiring a
+single data wire to connect and the latter also needing a wire for the clock
+signal.
+
```
Comment (nielsnl68) [inline components/light/fastled.rst:40]:
```suggestion

.. warning::

    The ``fastled_spi`` and the ``fastled_clocless`` light platform are now deprecated.
    Use the new, unified ``fastled`` light platform, instead.
```
My suggestion is to move the warning from the bottom of this document to here and remove the section about the ``fastled_spi`` and the ``fastled_clocless``.

**Review by nielsnl68 — CHANGES_REQUESTED** [review]
Thanks for changing the document for me. I have some small changing suggestions added below/

**Comment by bibach** [comment]
@nielsnl68 Good catch about the platform names in the examples.  I meant to update those after copying them from the other sections.

I initially hesitated on removing the existing material, as I wasn't sure if you intended to deprecate the other components.  Since it seems you're good to proceed with that, I took it out.

**Comment by nielsnl68** [comment]
Still waiting for the esphome-PR to be merged

**Comment by nielsnl68** [comment]
'ping

**Pull request closed without merging**